src/posts/uploads.js

```diff
@@ -51,14 +51,14 @@ module.exports = function (Posts) {
 		});
 	};
 
-	Posts.uploads.list = async function (pid, callback) {
+	Posts.uploads.list = function (pid, callback) {
 		// Returns array of this post's uploads
 		db.getSortedSetRange('post:' + pid + ':uploads', 0, -1, callback);
 	};
 
 	Posts.uploads.listWithSizes = async function (pid) {
 		const paths = await Posts.async.uploads.list(pid);
-		const sizes = await db.getObjects(paths.map(path => 'upload:' + md5(path)));
+		const sizes = await db.async.getObjects(paths.map(path => 'upload:' + md5(path)));
```

Comment:
@barisusakli I experienced an issue here. If at top of file I `require('../database').async`, then later on in the file, `console.log(db);` was `undefined` if called from inside `Posts.upload.list`.

I ended up fixing it by doing `db.async` when used inside the file instead.

Just to give you some context, the upgrade script I was writing was calling `Posts.uploads.list`, and erroring out as `db` was `undefined`.